app/src/main/java/de/tum/in/tumcampusapp/component/ui/transportation/DepartureView.kt

```diff
@@ -0,0 +1,185 @@
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.animation.AccelerateInterpolator
+import android.view.animation.AnimationUtils
+import android.widget.LinearLayout
+import android.widget.TextSwitcher
+import android.widget.TextView
+import de.tum.`in`.tumcampusapp.R
+import org.joda.time.DateTime
+import org.joda.time.Seconds
+import java.util.*
+
+/**
+ * Custom view that shows a departure.
+ * Holds an icon of the subway public transfer line, the line name and an animated
+ * automatically down counting departure time
+ */
+class DepartureView
+/**
+ * Constructor for DepartureView
+ *
+ * @param context Context
+ * @param big     Whether the departure should use a thin or a big line
+ */
+@JvmOverloads constructor(context: Context, private val big: Boolean = false) : LinearLayout(context) {
+    private val symbolView: TextView by lazy { findViewById<TextView>(R.id.line_symbol) }
+    private val lineView: TextView by lazy { findViewById<TextView>(R.id.nameTextView) }
+    private val timeSwitcher: TextSwitcher by lazy { findViewById<TextSwitcher>(R.id.line_switcher) }
+    private val mHandler: Handler
+    private var valueAnimator: ValueAnimator? = null
+    private var departureTime: DateTime? = null
+
+    val symbol: String
+        get() = symbolView.text.toString()
+
+    init {
+
+        orientation = HORIZONTAL
+        gravity = Gravity.CENTER_VERTICAL
+
+        val inflater = LayoutInflater.from(context)
+        if (big) {
+            inflater.inflate(R.layout.departure_line_big, this, true)
+        } else {
+            inflater.inflate(R.layout.departure_line_small, this, true)
+        }
+
+        // Declare the in and out animations and initialize them
+        val `in` = AnimationUtils.loadAnimation(getContext(), android.R.anim.slide_in_left)
+        val out = AnimationUtils.loadAnimation(getContext(), android.R.anim.slide_out_right)
+
+        // set the animation type of textSwitcher
+        timeSwitcher.inAnimation = `in`
+        timeSwitcher.outAnimation = out
+
+        mHandler = Handler()
+    }
+
+    /**
+     * Sets the line symbol name
+     *
+     * @param symbol Symbol e.g. U6, S1, T14
+     */
+    fun setSymbol(symbol: String, highlight: Boolean) {
+        val mvvSymbol = MVVSymbol(symbol)
+        symbolView.setTextColor(mvvSymbol.textColor)
+        symbolView.text = symbol
+        symbolView.backgroundTintList = ColorStateList.valueOf(mvvSymbol.backgroundColor)
+
+        if (highlight) {
+            if (big) {
+                setBackgroundColor(mvvSymbol.backgroundColor)
+                lineView.setTextColor(Color.WHITE)
+                for (index in 0 until timeSwitcher.childCount) {
+                    val tw = timeSwitcher.getChildAt(index) as TextView
+                    tw.setTextColor(Color.WHITE)
+                }
+            } else {
+                setBackgroundColor(0x20ffffff and mvvSymbol.backgroundColor)
```

Comment:
That’s not good. I assume this code adds some transparency to `mvvSymbol.backgroundColor`, but that should be done in an easier-to-understand way. Maybe you can encapsulate it in a method/backed property on `MVVSymbol`?